pudb/debugger.py

```diff
@@ -214,6 +223,32 @@ def __init__(self, stdin=None, stdout=None, term_size=None, steal_output=False,
         # Okay, now we have a debugger
         self._current_debugger.append(self)
 
+    def reset(self):
+        """Set values of attributes as ready to start debugging.
+
+        Override from Bdb.reset()
+
+        When pytest starts a postmortem analysis, but the debugger is already active,
+        calling .reset() in src/_pytest/debugging.py::post_mortem
+        causes the self.stopframe to be set to None.
+        This pauses the debugger somewhere in the source code of the debugger. See #67
+
+        We detect using _pytest_postmortem that this is the case and do not set the
+        stopframe to None then.
+
+        Related #607, #52
+        """
+        import linecache
+        linecache.checkcache()
+        self.botframe = None
+        if not self._pytest_postmortem:
+            self.stopframe = None
+        else:
+            self._pytest_postmortem = False
+        self.returnframe = None
+        self.quitting = False
+        self.stoplineno = 0
```

Comment:
Would it be possible to call into `Bdb.reset` for some of this? We're already copy-pasting too much of Bdb's guts into pudb... (If not, add a comment why not.)

Reply:
When calling Bdb.reset the debugger stops around there. `self.stopframe = None` is the offending line. Calling Bdb.reset is a no-go. It is the exact same problem for #67 where also code is copies from Bdb in here. The problem now is that pytest is calling .reset, so that is why the override is needed and to check for the pytest postmortem state. I can add a link to the source of pytest as well, but I believe the docstring of this method explain already enough.

Do you think there needs to be more information in the docstring of this method to explain why it needs to be overriden?